src/Sulu/Component/Content/Document/Subscriber/WebspaceSubscriber.php

```diff
@@ -49,13 +62,39 @@ public function supports($document)
     }
 
     /**
-     * @param HydrateEvent $event
+     * @param AbstractMappingEvent|HydrateEvent $event
+     * @throws \Sulu\Component\DocumentManager\Exception\DocumentManagerException
      */
     public function doHydrate(AbstractMappingEvent $event)
     {
         $document = $event->getDocument();
-        $webspace = $this->inspector->getWebspace($document);
-        $event->getAccessor()->set('webspaceName', $webspace);
+        $webspaceName = $this->inspector->getWebspace($document);
+        $event->getAccessor()->set('webspaceName', $webspaceName);
+
+        if ($document instanceof StructureBehavior && $document->getStructureType() === null) {
+            $this->setDefaultStructureType($event, $document, $webspaceName);
+        }
```

Comment:
yeah, currently there is no mapping for required (with the pending PRs on the document manager this would be easy to implement however), but I just meant that it should be a requirement when saving a Document that contains a structure.

What is the use case for saving a structure without a structure type? If you needed to set data on the structure, you would need to know its type before doing so I think.